### PR TITLE
[CodeCompletion] Propagate completion status of parseGenericArguments()

### PR DESCRIPTION
**Pull request by rintaro**
To properly handle code-completion in generic argument list.

rdar://problem/39312854

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseExpr.cpp [hunk 1]
```diff
@@ -1207,9 +1207,9 @@ Parser::parseExprPostfixSuffix(ParserResult<Expr> Result, bool isExprBasic,
       if (canParseAsGenericArgumentList()) {
         SmallVector<TypeRepr *, 8> args;
         SourceLoc LAngleLoc, RAngleLoc;
-        if (parseGenericArguments(args, LAngleLoc, RAngleLoc)) {
+        auto argStat = parseGenericArguments(args, LAngleLoc, RAngleLoc);
```
Comment (benlangmuir) [inline lib/Parse/ParseExpr.cpp]: Shouldn't this status be incorporated into the ParserResult about to be returned?

File: lib/Parse/ParseExpr.cpp [hunk 2]
```diff
@@ -2259,9 +2259,9 @@ Expr *Parser::parseExprIdentifier() {
   if (canParseAsGenericArgumentList()) {
     SyntaxContext->createNodeInPlace(SyntaxKind::IdentifierExpr);
     SyntaxContext->setCreateSyntax(SyntaxKind::SpecializeExpr);
-    if (parseGenericArguments(args, LAngleLoc, RAngleLoc)) {
+    auto argStat = parseGenericArguments(args, LAngleLoc, RAngleLoc);
```
Comment (benlangmuir) [inline lib/Parse/ParseExpr.cpp]: Same here.

File: lib/Parse/ParseExpr.cpp (hunk 1 above)
Comment (rintaro) [reply lib/Parse/ParseExpr.cpp]: Currently, that's unreachable because `canParseAsGenericArgumentList()` fails if CC token is in it.

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**